### ISSUE TITLE
[csharp] generating blank objects when type is string

### ISSUE THREAD

**Issue by alexl0gan** [swagger-api/swagger-codegen#7707]
##### Description

When defining the scheme for a return value that has type string, codegen generates a blank object that has no properties, therefore is unusable. In the example below, the object generated for Item, isn't a string, its an object that has no properties. 

##### Swagger-codegen version

2.2.3

##### Swagger declaration file content or url

```yaml
/some-endpoint/{id}:
    get:
      summary: An endpoint
      produces:
        - text/uri-list
      parameters:
        - name: id
          in: path
          type: string
      responses:
        200:
          description: An Item.
          schema:
            $ref: '#/definitions/Item'

Item:
    type: string
```

Object generated by this:

```csharp
using System;
using System.Linq;
using System.IO;
using System.Text;
using System.Text.RegularExpressions;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Runtime.Serialization;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using System.ComponentModel.DataAnnotations;
using SwaggerDateConverter = mds.Client.SwaggerDateConverter;

namespace mds.Model
{
    /// <summary>
    /// An item
    /// </summary>
    [DataContract]
    public partial class Item :  IEquatable<Item>, IValidatableObject
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="Item" /> class.
        /// </summary>
        [JsonConstructorAttribute]
        public Item()
        {
        }
        
        /// <summary>
        /// Returns the string presentation of the object
        /// </summary>
        /// <returns>String presentation of the object</returns>
        public override string ToString()
        {
            var sb = new StringBuilder();
            sb.Append("class Item {\n");
            sb.Append("}\n");
            return sb.ToString();
        }
  
        /// <summary>
        /// Returns the JSON string presentation of the object
        /// </summary>
        /// <returns>JSON string presentation of the object</returns>
        public string ToJson()
        {
            return JsonConvert.SerializeObject(this, Formatting.Indented);
        }

        /// <summary>
        /// Returns true if objects are equal
        /// </summary>
        /// <param name="input">Object to be compared</param>
        /// <returns>Boolean</returns>
        public override bool Equals(object input)
        {
            return this.Equals(input as ResourcePath);
        }

        /// <summary>
        /// Returns true if ResourcePath instances are equal
        /// </summary>
        /// <param name="input">Instance of ResourcePath to be compared</param>
        /// <returns>Boolean</returns>
        public bool Equals(ResourcePath input)
        {
            if (input == null)
                return false;

            return false;
        }

        /// <summary>
        /// Gets the hash code
        /// </summary>
        /// <returns>Hash code</returns>
        public override int GetHashCode()
        {
            unchecked // Overflow is fine, just wrap
            {
                int hashCode = 41;
                return hashCode;
            }
        }

        /// <summary>
        /// To validate all properties of the instance
        /// </summary>
        /// <param name="validationContext">Validation context</param>
        /// <returns>Validation Result</returns>
        IEnumerable<System.ComponentModel.DataAnnotations.ValidationResult> IValidatableObject.Validate(ValidationContext validationContext)
        {
            yield break;
        }
    }

}
```

**Comment by chris-hatton** [comment]
+1  We are affected by this and are so far working around it by manually replacing instances of the 'Dummy' object with string.

**Comment by alexl0gan** [comment]
This used to be an issue in Java but it was fixed between 2.2.2 and 2.3.0

**Comment by weiffert-bhs** [comment]
I'm getting this same thing with version 2.4.5

**Comment by CollyMcK** [comment]
having this issue as well using version 3.0.19..using the workaround of commenting out the class and then changing the property to a string.

Is there a better workaround even?

**Comment by alexl0gan** [comment]
@CollyMcK our workaround was to ban the api team from doing this 😆 

Surprised this still happens 2.5 years later...

**Comment by CollyMcK** [comment]
@alexl0gan - you mean the team writing the YAML file - what way should it be changed to avoid the issue?

**Comment by alexl0gan** [comment]
@CollyMcK yep the team writing the yaml. Just returning a string from an endpoint is not great REST either. 

I haven't touched swagger codegen for years now. We wrote our own generators and it worked much better.

**Comment by CollyMcK** [comment]
@alexl0gan what should the team defining the yaml be using instead of string..the underlying class/object as well.

Take it you didn't open source your own generator?

**Comment by gigaplex** [comment]
Why was this closed if it's still an issue? And string is a fundamental data type for interoperability, you can't just explicitly ban its usage.

**Comment by toddsay** [comment]
I was encountering this same problem in swagger-codegen 2.4.13, and found it to be fixed in 2.4.14 due to https://github.com/swagger-api/swagger-codegen/pull/10190.

As for the comment questioning the banning of usage: Presumably it was the use of separate property definitions for primitive types that was being banned (so, just define the string inline instead of creating a separate type). But it's no longer an issue with the newer generator.
